accounts/abi/method.go

```diff
@@ -90,54 +90,51 @@ func (method Method) tupleUnpack(v interface{}, output []byte) error {
 	var (
 		value = valueOf.Elem()
 		typ   = value.Type()
+		kind  = value.Kind()
 	)
+	if err := requireUnpackKind(value, typ, kind, m.Outputs, false); err != nil {
+		return err
+	}
 
 	j := 0
-	for i := 0; i < len(method.Outputs); i++ {
-		toUnpack := method.Outputs[i]
-		if toUnpack.Type.T == ArrayTy {
+	for i, o := range m.Outputs {
```

Comment:
can we please use a more descriptive name than `o`?

Reply:
This variable is used very often, and doesn't have a big context. But If you want I can change to something else.